ArjunaJTA/jms/src/main/java/org/jboss/narayana/jta/jms/JmsXAResourceRecoveryHelper.java

```diff
@@ -177,4 +168,10 @@ private void disconnect() throws XAException {
         }
     }
 
+    private void assertIfRecoveryScanWasStarted() throws XAException {
+        if (delegate == null) {
+            throw new XAException("Recovery scan has to be started");
```

Comment:
OK. What about the use of XA codes instead of using the (unspecified) default code in the XAException constructor?